### PR TITLE
Mass Asset Import

### LINKED ISSUE #480: [FEATURE] Import Asset feature
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

When moving to AdamRMS, it is challenging to import an existing list of assets, without having to manually add them all

### Describe the solution you'd like

An Asset Import feature that automatically creates Assets and Asset Types based on a csv

### Describe alternatives you've considered

Manually entering each asset
A DB upload - challenging as have to import asset types, then get type IDs, then import assets

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cherry-john**
By submitting a PR for this repository you accept the contributor license agreement. 

### Description

Imports assets through a .csv upload from the Server Admin pages.

Functionality is restricted to Server Admins as this can affect the RMS on a large scale.

image
_A Successful Upload_

image
_An Upload with Errors_

### References

Closes #480 

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in the documentation repo
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: html/admin/api/server/assetImport.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        }
+        
+        //Asset Category
+        $DBLIB->where("assetCategories_id", $row[7]);
+        $DBLIB->where("(assetCategories.instances_id = ? or assetCategories.instances_id IS NULL)", [$_POST['instances_id']]);
+        $DBLIB->where("assetCategories.assetCategories_deleted", 0);
+        $assetCategory = $DBLIB->getOne("assetCategories", ["assetCategories.assetCategories_id"]);
+        if (!$assetCategory) {
+            //Asset Category not found
+            //This is the one thing we can't just create with data from the CSV
```
Comment (Jbithell) [inline html/admin/api/server/assetImport.php:101]: Could we use the category name instead rahter than the ID?

File: html/admin/api/server/assetImport.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        if ($assetType['assetTypes_id']) array_push($createdAssetTypes, $assetType);
+        else array_push($failedAssets, ["tag" => $row[9], "reason" => "Error creating Asset Type"]);
+
+    }
+
+    //Asset Tag
+    if (!isset($row[9]) or $row[9] == null) $row[9] = generateNewTag();
+    else {
+        //check Tag is not going to be duplicated
+        $DBLIB->where("assets.instances_id",$AUTH->data['instance']['instances_id']);
```
Comment (Jbithell) [inline html/admin/api/server/assetImport.php:139]: Is this not the same as line 75?

File: html/admin/public/templates/asset-import-template.csv
Comment (Jbithell) [inline html/admin/public/templates/asset-import-template.csv:1]: I'd move this file to /assets/ as /public/ is for the public sites feature

**Review by Jbithell — CHANGES_REQUESTED** [review]
Think this looks really good, few tiny things

File: html/admin/api/server/assetImport.php (hunk 2 above)
Comment (cherry-john) [reply html/admin/api/server/assetImport.php:139]: This checks for the asset not asset type, so you can't import a duplicate asset tag

**Review by Jbithell — APPROVED** [review]

**Pull request merged**